csharp/src/Ice/Endpoint.cs

```diff
@@ -117,30 +132,38 @@ public override int GetHashCode()
             return hash.ToHashCode();
         }
 
-        /// <summary>Converts the endpoint into a string, using the old string format.</summary>
-        // TODO: add a parameter to select the format?
-        public override string ToString() => $"{TransportName}{OptionsToString()}";
-
-        /// <summary>Converts all the options to a string with a leading empty space character, using the old format
-        /// for stringified proxies/endpoints.</summary>
-        public abstract string OptionsToString();
+        /// <summary>Converts the endpoint into a string. The format of thi string depends on the protocol: either
+        /// ice1 format (for ice1) or URI format (for ice2 and up).</summary>
+        public override string ToString()
+        {
+            if (Protocol == Protocol.Ice1)
+            {
+                var sb = new StringBuilder(Scheme);
+                AppendOptions(sb, ' '); // option separator is not used with ice1
+                return sb.ToString();
+            }
+            else
+            {
+                var sb = new StringBuilder();
+                sb.AppendEndpoint(this);
+                return sb.ToString();
+            }
```

Comment:
AppendOptions is a virtual method implemented by derived classes, it can't be an extension. And there is no point in wrapping it in another extension method.